lib/compiler.js

```diff
@@ -94,18 +95,13 @@ Compiler.prototype.generate = function generate(code, options) {
   if (!options.to) options.to = process.cwd();
   var file = new SourceMapFile(options.to, { sourceMap: sourceMap });
 
-  code = [
-    fnToString(code) + ';',
-    'oninit(function(exports, context) {',
-      'var BEMContext = exports.BEMContext || context.BEMContext;',
-      // Provides third-party libraries from different modular systems
-      'BEMContext.prototype.require = function(lib) {',
-       'return this._libs[lib];',
-      '};',
-    '});'
-  ].join('');
+  code = fnToString(code);
 
   var deps = getDeps(options.requires);
+  var bemxjstOpts = utils.extend({}, options);
```

Comment:
any reason to use custom `extend` instead of `Object.assign()` here?

Reply:
@tadatuta replaced. Thanks!

Reply:
@tadatuta let’s drop all not actual LTS Node.js support in next major version.

Reply:
>  Ah, we still support node 0.12 =)

🙀

Reply:
@sbmaxx well, because technicaly it’s a major release =)